jni/include/faiss_index_service.h

```diff
@@ -31,8 +31,50 @@ namespace faiss_wrapper {
 class IndexService {
 public:
     IndexService(std::unique_ptr<FaissMethods> faissMethods);
-    //TODO Remove dependency on JNIUtilInterface and JNIEnv
-    //TODO Reduce the number of parameters
+    /**
```

Comment:
Right now functions are taking 8 parameters , Ideally we should not increase more than 2. This is not scalable , 

Shall we wrap it like this , It makes api more cleaner.
```

struct IndexConfiguration {
    knn_jni::JNIUtilInterface* jniUtil;
    JNIEnv* env;
    faiss::MetricType metric;
    std::string indexDescription;
    int dim;
    int numVectors;
    int threadCount;
    std::unordered_map<std::string, std::string> parameters;
};

class IndexService {
public:
    virtual jlong initIndex(const IndexConfiguration& config);
};
```

Reply:
This is a nice way to do it. It would require some work, though, since we need a solution that doesn't require remaking `parameters` on each InsertToIndex. In order to do this, then, we would probably need InitIndex to return the address to the IndexConfiguration, then use the address as the input for subsequent functions. I will look into this.

Reply:
Sure . let me know if you need any help , we can write together

Reply:
This is to be resolved in a later PR when the JNI layer fixes are implemented 👍